trio/tests/test_exports.py

```diff
@@ -18,3 +23,21 @@ def test_core_is_properly_reexported():
                 found += 1
         print(symbol, found)
         assert found == 1
+
+
+def test_pylint_sees_all_non_underscore_symbols_in_namespace():
+    # Test pylints ast to contain the same content as dir(trio)
+    linter = PyLinter()
+    ast_set = set(linter.get_ast(trio.__file__, 'trio'))
+    trio_set = set([symbol for symbol in dir(trio) if symbol[0] != '_'])
+    trio_set.remove('tests')
```

Comment:
Probably worthwhile to factor out some of the namespace cleanup code from the two functions – I think in both cases we want to test something like: "the two sets are equivalent, if you ignore names that `.startswith("_")`, `"tests"`, and `"testing"`"?

Reply:
Ok, I'll work myself through all  the comments. Will take some time as this is all new stuff.

Reply:
No worries, take your time, and don't be afraid to ask for help if you get stuck :-)